sway-core/src/transform/to_parsed_lang/convert_parse_tree.rs

```diff
@@ -2044,6 +2045,199 @@ fn expr_func_app_to_expression_kind(
                 },
             ));
         }
+        // "__dbg(arg)" in debug becomes "{
+        //      let mut f = Formatter { };
+        //      f.print_str("[{current_file}:{current_line}:{current_col}] = ");
+        //      let arg = arg;
+        //      arg.fmt(f);
+        //      arg
+        // }"
+        Some(Intrinsic::Dbg)
+            if context.is_dbg_generation_full() && last.is_none() && !is_relative_to_root =>
+        {
+            if arguments.len() > 1 {
+                return Err(handler.emit_err(CompileError::IntrinsicIncorrectNumArgs {
+                    name: Intrinsic::Dbg.to_string(),
+                    expected: 1,
+                    span,
+                }));
+            }
+
+            let f_id: String = format!("f_{}", context.next_for_unique_suffix());
+            let f_ident = BaseIdent::new_no_span(f_id.to_string());
+
+            let f_tid = engines.te().new_unknown();
+            let f_decl_pid = engines.pe().insert(VariableDeclaration {
+                name: f_ident.clone(),
+                type_ascription: GenericArgument::Type(GenericTypeArgument {
+                    type_id: f_tid,
+                    initial_type_id: f_tid,
+                    span: span.clone(),
+                    call_path_tree: None,
+                }),
+                body: Expression {
+                    kind: ExpressionKind::Struct(Box::new(StructExpression {
+                        resolved_call_path_binding: None,
+                        call_path_binding: TypeBinding {
+                            inner: CallPath {
+                                prefixes: vec![],
+                                suffix: BaseIdent::new_no_span("Formatter".into()),
+                                callpath_type: CallPathType::Ambiguous,
+                            },
+                            type_arguments: TypeArgs::Regular(vec![]),
+                            span: span.clone(),
+                        },
+                        fields: vec![],
+                    })),
+                    span: span.clone(),
+                },
+                is_mutable: true,
+            });
+
+            let current_file = span
+                .source_id()
+                .map(|x| engines.se().get_path(x))
+                .unwrap_or_default();
```

Comment:
This looks dangerous to me. What is `PathBuf::default()`? I understand that in all the `unwrap()`s in this implementation we expect things to exist. Why not only `unwrap()` here as well? It should never fail and if it does failing is the right thing compared to have some default file that is not related to the `source_id` and then failing in the `unwrap()` below.

Reply:
Because we need to support `Span::dummy()`, and in this case it is better to not print the file name than panicking the compiler.

Reply:
The idea was to support `Span::dummy()`. I am treating it more explicitly now.